### DISCUSSION TITLE
Difference between glyph 'advance' and 'width'.

### DISCUSSION THREAD

**Discussion by kaiwk** [harfbuzz/harfbuzz#2892] [Q&A] [answered]
<img width="1225" alt="Screen Shot 2021-03-09 at 10 47 37 AM" src="https://user-images.githubusercontent.com/10190397/110423696-7fc96e80-80dc-11eb-97df-e4ab65a8d2af.png">

If I understand correctly, in right side picture (the font is Zapfino) one should use the sum of glyphs' width rather than advance as the text's bounds.

I want to know when should I use advance or width, for example,  should I use glyph advance for line wrap?

**Comment by behdad** [comment, marked as answer]
> I want to know when should I use advance or width, for example, should I use glyph advance for line wrap?

Glyph advance is used to know where to place next glyph. The width encodes the measured size of the ink. Those are completely different and can vary as you see.

The bounds of the text is NOT done by adding the widths (or advances), but by combining the bounds of each glyph at their position.
